miio/vacuum_cli.py

```diff
@@ -314,17 +314,31 @@ def dnd(vac: miio.Vacuum, cmd: str,
 
 
 @cli.command()
-@click.argument('speed', type=int, required=False)
+@click.argument('speed', type=str, required=False)
 @pass_dev
 def fanspeed(vac: miio.Vacuum, speed):
     """Query and adjust the fan speed."""
+    from miio.vacuum import FanSpeed
     if speed:
         click.echo("Setting fan speed to %s" % speed)
+        try:
```

Comment:
I think it makes sense first to check for the preset name, and then fall back to int conversion. Furthermore, if the conversion is not possible, and the exception should be captured and reported back.